File: backend/app/schemas/outreach.py
import uuid
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional, List, Dict, Any
from enum import Enum

class OutreachMode(str, Enum):
    email = "email"
    whatsapp = "whatsapp"
    phone_script = "phone_script"

class OutreachLanguage(str, Enum):
    de = "de" # German (DIN 5008)
    en = "en" # International Business English

class OutreachRequest(BaseModel):
    buyer_id: str
    mode: OutreachMode = OutreachMode.email
    language: OutreachLanguage = OutreachLanguage.en
    tone: str = "Professional" # Professional, Direct, Technical, Relationship
    contact_name: Optional[str] = None

class CompliancePackDoc(BaseModel):
    doc_id: str
    title: str
    document_type: str # lab_test, eudr_due_diligence, environmental_audit, reach_declaration
    issuer: str
    verified_date: str
    file_format: str = "PDF"

class CompliancePackResponse(BaseModel):
    bundle_id: str
    buyer_id: str
    buyer_name: str
    exporter_name: str
    documents: List[CompliancePackDoc]
    total_documents: int
    generated_at: str
    download_url: str

class OutreachResponse(BaseModel):
    action_id: str
    buyer_id: str
    buyer_name: str
    contact_name: str
    contact_title: str
    mode: OutreachMode
    language: OutreachLanguage
    tone: str
    subject: str
    body: str
    why_matches_you: List[str]
    compliance_pack_docs: List[str]
    status: str

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/outreach.ts
import { fetchApi } from './client';

export type OutreachMode = 'email' | 'whatsapp' | 'phone_script';
export type OutreachLanguage = 'de' | 'en';

export interface CompliancePackDoc {
  doc_id: string;
  title: string;
  document_type: string;
  issuer: string;
  verified_date: string;
  file_format: string;
}

export interface CompliancePackResponse {
  bundle_id: string;
  buyer_id: string;
  buyer_name: string;
  exporter_name: string;
  documents: CompliancePackDoc[];
  total_documents: number;
  generated_at: string;
  download_url: string;
}

export interface OutreachResponse {
  action_id: string;
  buyer_id: string;
  buyer_name: string;
  contact_name: string;
  contact_title: string;
  mode: OutreachMode;
  language: OutreachLanguage;
  tone: string;
  subject: string;
  body: string;
  why_matches_you: string[];
  compliance_pack_docs: string[];
  status: string;
}

export const generateOutreachApi = (payload: {
  buyer_id: string;
  mode?: OutreachMode;
  language?: OutreachLanguage;
  tone?: string;
  contact_name?: string;
}) =>
  fetchApi<OutreachResponse>('/api/v1/outreach', {
    method: 'POST',
    body: JSON.stringify(payload),
  });

export const getCompliancePackApi = (buyerId: string) =>
  fetchApi<CompliancePackResponse>(`/api/v1/outreach/compliance-pack/${buyerId}`);
